### PR TITLE
Add settings and compatibility with moreblocks and stairs

### LINKED ISSUE #57: Open question about xdecor elements and related standalone mods.
Several xdecor elements exist as part of xdecor and as standalone mod.
- https://github.com/minetest-mods/realchess
- https://github.com/minetest-mods/enchanting
- https://github.com/minetest-mods/workbench
- https://github.com/minetest-mods/craftguide

To reduce maintenance and development efforts, it would be great if we could deduplicate these.
For example by removing them from xdecor or by closing the standalone mod.

This decision could of course be made for each element individually.

What do **you** think?
Which elements would you like to use separate of xdecor, which would you miss in xdecor?

### PR DESCRIPTION
**Pull request by Jat15**
If workbench disabling.
And setting for diasabling:
- Chess
- Cooking
- Enchanting
- Hive
- Itemframe
- Mailbox
- Mechanisms
- Rope
- Workbench

By default all active.

Fixes #57

### CODE REVIEW

File: settingtypes.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#For desabling a element in xdecor.
```
Comment (kilbith) [inline settingtypes.txt]: Mispelling of "disabling".

File: settingtypes.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+disable_xdecor_chess 		(Diasable element Chess) 		bool false
```
Comment (kilbith) [inline settingtypes.txt]: Mispelling of "Disable".

Comment (Jat15) [reply settingtypes.txt]: Fix

File: settingtypes.txt (hunk 1 above)
Comment (Jat15) [reply settingtypes.txt]: Fix

File: handlers/registration.lua [hunk 3]
```diff
@@ -88,4 +118,26 @@ function xdecor.register(name, def)
 	end
 
 	minetest.register_node("xdecor:"..name, def)
+	
+	if minetest.settings:get_bool("disable_xdecor_workbench") and (minetest.get_modpath("moreblocks") or minetest.get_modpath("stairs")) then
```
[the remark this answers is not in the document]
Comment (Jat15) [reply handlers/registration.lua]: Fix

File: handlers/registration.lua [hunk 4]
```diff
@@ -43,6 +43,36 @@ local default_can_dig = function(pos)
 end
 
 function xdecor.register(name, def)
+	local function xdecor_stairs_alternatif(nodename, def)
```
[the remark this answers is not in the document]
Comment (Jat15) [reply handlers/registration.lua]: Fix

File: handlers/registration.lua [hunk 5]
```diff
@@ -43,6 +43,36 @@ local default_can_dig = function(pos)
...
+		local mod = string.match (nodename,"(.+):")
+		local name = string.match (nodename,":(.+)")
```
[the remark this answers is not in the document]
Comment (Jat15) [reply handlers/registration.lua]: Fix

File: handlers/registration.lua [hunk 6]
```diff
@@ -43,6 +43,36 @@ local default_can_dig = function(pos)
...
+		for i,groupname in ipairs(def.groups) do
+			if groupname == not("cracky") or not("choppy") or not("flammable") or not("crumbly") or not("snappy") then
+				table.remove(def[groups][groupname])
```
[the remark this answers is not in the document]
Comment (Jat15) [reply handlers/registration.lua]: fix

File: handlers/registration.lua [hunk 7]
```diff
@@ -43,6 +43,36 @@ local default_can_dig = function(pos)
 end
 
 function xdecor.register(name, def)
+	local function xdecor_stairs_alternatif(nodename, def)
+		local mod = string.match (nodename,"(.+):")
+		local name = string.match (nodename,":(.+)")
+		for i,groupname in ipairs(def.groups) do
+			if groupname == not("cracky") or not("choppy") or not("flammable") or not("crumbly") or not("snappy") then
```
[the remark this answers is not in the document]
Comment (Jat15) [reply handlers/registration.lua]: Big error ! 
Fix

File: handlers/registration.lua [hunk 8]
```diff
@@ -88,4 +118,26 @@ function xdecor.register(name, def)
...
+		   not def.allow_metadata_inventory_take and
+		   not (def.groups.not_in_creative_inventory == 1) and
+		   not (def.groups.not_cuttable == 1) and
+		   not def.groups.wool and
+		   (def.tiles and type(def.tiles[1]) == "string" and not
+			def.tiles[1]:find("default_mineral")) and
+		   not def.mesecons and
+		   def.description and
+		   def.description ~= "" and
+		   def.light_source == 0
```
[the remark this answers is not in the document]
Comment (Jat15) [reply handlers/registration.lua]: Mhhhh it is copy a code workbench.
I fix both ?

[the remark this answers is not in the document]
Comment (kilbith) [reply handlers/registration.lua]: No. I mean, you duplicated this long condition already, so move it to a function which return a boolean.

File: handlers/registration.lua [hunk 9]
```diff
@@ -43,6 +43,40 @@ local default_can_dig = function(pos)
 end
 
 function xdecor.register(name, def)
+	local function xdecor_stairs_alternative(nodename, def)
+		local mod, name = nodename:match("(.*):(.*)")
+		for groupname,value in pairs(def.groups) do
```
Comment (kilbith) [inline handlers/registration.lua]: Missing space after comma.

File: handlers/registration.lua [hunk 10]
```diff
@@ -43,6 +43,40 @@ local default_can_dig = function(pos)
...
+			if	groupname ~= "cracky" and
+				groupname ~= "choppy" and
+				groupname ~="flammable" and
+				groupname ~="crumbly" and
+				groupname ~="snappy"
```
Comment (kilbith) [inline handlers/registration.lua]: Various style issues here and above.

**Comment by kilbith** [comment]
I will likely have to fix all the style issues on my own I think.

**Comment by Jat15** [comment]
More @kilbith ?
And sorry for my english is not my language !

File: handlers/helpers.lua [hunk 11]
```diff
@@ -29,3 +29,27 @@ function xdecor.tablecopy(T)
...
+		not def.mesecons and
+		def.description and
+		def.description ~= "" and
+		def.light_source == 0
+	then
+		return true
+	else
+		return false
+	end
+end
```
Comment (kilbith) [inline handlers/helpers.lua]: You could just do that : 

```Lua
function()
     return [statements]
end
```

**Comment by Jat15** [comment]
Wait fix Function and bug...

**Comment by Jat15** [comment]
Finish for me.

File: handlers/helpers.lua [hunk 12]
```diff
@@ -29,3 +29,25 @@ function xdecor.tablecopy(T)
...
+		(def.tiles and type(def.tiles[1]) == "string" and not
+		def.tiles[1]:find("default_mineral")) and
+		not def.mesecons and
+		def.description and
+		def.description ~= "" and
+		def.light_source == 0
+	then
+		return true
+	end
+end
```
Comment (kilbith) [inline handlers/helpers.lua]: You have not understood what I wrote. Just a simple return for all these statements is sufficient.

Please learn Lua better.

**Comment by Jat15** [comment]
Something else ?

**Comment by Jat15** [comment]
@kilbith ?

**Comment by sofar** [comment]
conflicts

**Comment by Jat15** [comment]
@sofar  rebase

**Comment by kilbith** [comment]
Can you rebase please?

**Pull request merged**